portfolio/src/main/client/src/util/data.ts

```diff
@@ -0,0 +1,58 @@
+/** A function that responds to a signal changing. */
+type Handler<T> = (value: T, oldValue?: T) => void;
+
+/**
+ * A function that unregisters a @see {Handler}.
+ *
+ * @returns `true` if that handler was registered, `false` otherwise.
+ */
+type Cleanup = () => boolean;
+
+/**
+ * A data signal, an object which notifies subscribers when data is modified.
+ */
+export class DataSignal<T> {
```

Comment:
I'm not convinced you need to introduce this helper, and it probably hinders readability more than it helps it given the relatively small scope of this project.